### ISSUE TITLE
Problems with GPIO trigger using long wires

### ISSUE THREAD

**Issue by axel8417** [reuterbal/photobooth#16]
Hello,

First of all I want to say thank you, the new version of the photobooth is awesome! V1 was great, but this is even better!

But unforunatelly I have got a problem with the buzzer/trigger of my photobooth. It freezes the software at the "get ready ..." screen.

I use a raspberry pi 3b+ with the latest version of raspbian
python 3.5.3
Canon EOS 600D

My buzzer has a 5m long cable which connects it to the raspberry pi 3b+ I use. The error occured every time I wanted to take a picture, and the software stuck at the "get ready..." screen. But I could quit the software with esc, so it didn't freez up totally.
In Version 0.1 I had some problems with this buzzer triggering the photobooth randomly and the photobooth taking two pictures, so I put in a resistor which improved it a little. 
After I removed the resistor I could take at 0-3 pictures before the error occured. Removing the button and just connecting the pins with a short cable workd fine.

**Photobooth log:**
```

2018-06-24 08:43:04,409 - root - INFO - Initializing photobooth
2018-06-24 08:43:04,410 - root - INFO - Photobooth version: 0.3.dev16+g289928f
2018-06-24 08:43:04,413 - root - INFO - Reading config file "/home/pi/photobooth/photobooth/defaults.cfg"
2018-06-24 08:43:04,418 - root - INFO - Reading config file "photobooth.cfg"
2018-06-24 08:43:08,420 - root - INFO - Number of last existing file: 10
2018-06-24 08:43:08,424 - root - INFO - Saving assembled pictures as "2018-06-24/photoboothXXXXX.jpg"
2018-06-24 08:43:08,424 - root - INFO - Using python-gphoto2 bindings
2018-06-24 08:43:08,590 - gphoto2 - WARNING - (b'ptp_list_folder_eos [library.c:7342]') b'storage 0xffffffff, but handle 0x00000000?'
2018-06-24 08:43:08,595 - gphoto2 - WARNING - (b'ptp_usb_getresp [usb.c:465]') b'PTP_OC 0x9109 receiving resp failed: PTP General Error (0x2002)'
2018-06-24 08:43:08,596 - gphoto2 - WARNING - (b'ptp_list_folder_eos [library.c:7358]') b"'ptp_canon_eos_getobjectinfoex ( params, storageids.Storage[k], handle ? handle : 0xffffffff, 0x100000, &tmp, &nroftmp)' failed: PTP General Error (0x2002)"
2018-06-24 08:43:08,662 - root - INFO - Camera summary: Manufacturer: Canon Inc.
Model: Canon EOS 600D
  Version: 3-1.0.2
  Serial Number: xxxx
Vendor Extension ID: 0xb (2.0)

Capture Formats: JPEG
Display Formats: Association/Directory, Script, DPOF, MS AVI, MS Wave, JPEG, CRW, Unknown(b103), Unknown(bf02), Defined Type, Unknown(b104), Unknown(b105)

Device Capabilities:
	File Download, File Deletion, File Upload
	No Image Capture, No Open Capture, Canon EOS Capture, Canon EOS Shutter Button

Storage Devices Summary:

Device Property Summary:
Property 0xd402:(read only) (type=0xffff) 'Canon EOS 600D'
Property 0xd407:(read only) (type=0x6) 1
Property 0xd406:(readwrite) (type=0xffff) 'Unknown Initiator'
Property 0xd303:(read only) (type=0x2) 1
Battery Level(0x5001):(read only) (type=0x2) Enumeration [100,0,75,0,50] value: 100% (100)

2018-06-24 08:43:08,797 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,798 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,799 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,799 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,800 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,800 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,801 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,801 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,802 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,803 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,803 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,804 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,804 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,805 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,805 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,806 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,806 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,807 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,807 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,808 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,809 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,809 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,810 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,810 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,811 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,811 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,812 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,814 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,815 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,815 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,816 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,817 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,818 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,818 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,819 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,819 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,820 - gphoto2 - WARNING - (b'gp_widget_count_choices [gphoto2-widget.c:784]') b"Invalid parameters: '(widget->type == GP_WIDGET_RADIO) || (widget->type == GP_WIDGET_MENU)' is NULL/FALSE."
2018-06-24 08:43:08,820 - root - INFO - Camera configuration:
Camera Actions [actions]: 
    Synchronize camera date and time with PC [syncdatetime]: 0
    UI Lock [uilock]: 2
    Drive Canon DSLR Autofocus [autofocusdrive]: 0
    Drive Canon DSLR Manual focus [manualfocusdrive]: None (Near 1, Near 2, Near 3, None, Far 1, Far 2, Far 3, )
    Cancel Canon DSLR Autofocus [cancelautofocus]: 0
    Canon EOS Zoom [eoszoom]: 0
    Canon EOS Zoom Position [eoszoomposition]: 0,0
    Canon EOS Viewfinder [viewfinder]: 2
    Canon EOS Remote Release [eosremoterelease]: None (None, Press Half, Press Full, Release Half, Release Full, Immediate, Press 1, Press 2, Press 3, Release 1, Release 2, Release 3, )
    PTP Opcode [opcode]: 0x1001,0xparam1,0xparam2
Camera Settings [settings]: 
    Camera Date and Time [datetime]: 1529820244
    Quick Review Time [reviewtime]: 4 seconds (None, 2 seconds, 4 seconds, 8 seconds, Hold, )
    Camera Output [output]: PC (TFT, PC, TFT + PC, Setting 4, Setting 5, Setting 6, Setting 7, )
    Recording Destination [movierecordtarget]: Unknown value 0003 (Unknown value 0003, )
    EVF Mode [evfmode]: 1 (0, 1, )
    Owner Name [ownername]: 
    Artist [artist]: 
    Copyright [copyright]: 
    Custom Functions Ex [customfuncex]: c8,4,1,2c,3,101,1,0,103,1,0,10f,1,0,2,2c,3,201,1,0,202,1,0,203,1,0,3,20,2,50e,1,0,60f,1,0,4,38,4,701,1,0,704,1,0,811,1,0,80f,1,0,
    Focus Info [focusinfo]: eosversion=4,size=5184x3456,size2=5184x3456,points={{-1477,0,186,139},{-886,418,186,139},{-886,-418,186,139},{0,787,127,196},{0,0,231,238},{0,-787,127,196},{886,418,186,139},{886,-418,186,139},{1477,0,186,139}},select={},unknown={ff010000ffff}
    Auto Power Off [autopoweroff]: 0
    Depth of Field [depthoffield]: 0
    Capture Target [capturetarget]: Internal RAM (Internal RAM, Memory card, )
    Capture [capture]: 0
Camera Status Information [status]: 
    Serial Number [serialnumber]: 80b7725b79ad4d78bc9e1f7865dc46f9
    Camera Manufacturer [manufacturer]: Canon Inc.
    Camera Model [cameramodel]: Canon EOS 600D
    Device Version [deviceversion]: 3-1.0.2
    Vendor Extension [vendorextension]: None
    Camera Model [model]: 2147484294
    PTP Version [ptpversion]: 256
    Battery Level []: 100%
    Battery Level [batterylevel]: 100%
    Lens Name [lensname]: 18-200mm
    Serial Number [eosserialnumber]: 233076199374
    Shutter Counter [shuttercounter]: 5003
    Available Shots [availableshots]: 42101
Image Settings [imgsettings]: 
    Image Format [imageformat]: Large Fine JPEG (Large Fine JPEG, Large Normal JPEG, Medium Fine JPEG, Medium Normal JPEG, Small Fine JPEG, Small Normal JPEG, Smaller JPEG, Tiny JPEG, RAW + Large Fine JPEG, RAW, )
    Image Format SD [imageformatsd]: Large Fine JPEG (Large Fine JPEG, Large Normal JPEG, Medium Fine JPEG, Medium Normal JPEG, Small Fine JPEG, Small Normal JPEG, Smaller JPEG, Tiny JPEG, RAW + Large Fine JPEG, RAW, )
    ISO Speed [iso]: Auto (Auto, 100, 200, 400, 800, 1600, 3200, 6400, )
    WhiteBalance [whitebalance]: Flash (Auto, Daylight, Shadow, Cloudy, Tungsten, Fluorescent, Flash, Manual, )
    WhiteBalance Adjust A [whitebalanceadjusta]: 0 (-9, -8, -7, -6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, )
    WhiteBalance Adjust B [whitebalanceadjustb]: 0 (-9, -8, -7, -6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, )
    WhiteBalance X A [whitebalancexa]: 0
    WhiteBalance X B [whitebalancexb]: 0
    Color Space [colorspace]: sRGB (sRGB, AdobeRGB, )
Capture Settings [capturesettings]: 
    Exposure Compensation [exposurecompensation]: 0 (-5, -4.6, -4.3, -4, -3.6, -3.3, -3, -2.6, -2.3, -2, -1.6, -1.3, -1.0, -0.6, -0.3, 0, 0.3, 0.6, 1.0, 1.3, 1.6, 2, 2.3, 2.6, 3, 3.3, 3.6, 4, 4.3, 4.6, 5, )
    Focus Mode [focusmode]: One Shot (One Shot, )
    Canon Auto Exposure Mode [autoexposuremode]: P (P, TV, AV, Manual, Bulb, A_DEP, DEP, Custom, Lock, Green, Night Portrait, Sports, Portrait, Landscape, Closeup, Flash Off, )
    Drive Mode [drivemode]: Single (Single, Continuous, Timer 10 sec, Timer 2 sec, Unknown value 0007, )
    Picture Style [picturestyle]: Auto (Standard, Portrait, Landscape, Neutral, Faithful, Monochrome, Auto, User defined 1, User defined 2, User defined 3, )
    Aperture [aperture]: implicit auto (implicit auto, )
    Shutter Speed [shutterspeed]: auto (auto, )
    Bracket Mode [bracketmode]: 0
    Auto Exposure Bracketing [aeb]: off (off, +/- 1/3, +/- 2/3, +/- 1, +/- 1 1/3, +/- 1 2/3, +/- 2, )
Other PTP Device Properties [other]: 
    PTP Property 0xd402 [d402]: Canon EOS 600D
    PTP Property 0xd407 [d407]: 1
    PTP Property 0xd406 [d406]: Unknown Initiator
    PTP Property 0xd303 [d303]: 1
    Battery Level [5001]: 100 (100, 0, 75, 0, 50, )

2018-06-24 08:43:11,993 - root - INFO - Countdown with preview activated
2018-06-24 08:43:11,994 - root - INFO - GPIO enabled (lamp_pin=4, trigger_pin=23, exit_pin=24)
2018-06-24 08:43:13,564 - root - INFO - Photobooth triggered
2018-06-24 08:43:13,615 - root - ERROR - Unknown event received: triggered
```

It shows this in command line:
```
2018-06-24 08:50:20,002 - root - INFO - Photobooth triggered
2018-06-24 08:50:20,080 - root - ERROR - Unknown event received: triggered
Process CameraProcess-1:
Traceback (most recent call last):
  File "/home/pi/photobooth/photobooth/Photobooth.py", line 101, in recvEvent
    event_idx = expected.index(str(event))
ValueError: 'triggered' is not in list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/home/pi/photobooth/photobooth/main.py", line 79, in run
    status_code = self.run_camera()
  File "/home/pi/photobooth/photobooth/main.py", line 57, in run_camera
    return photobooth.run()
  File "/home/pi/photobooth/photobooth/Photobooth.py", line 147, in run
    self.trigger()
  File "/home/pi/photobooth/photobooth/Photobooth.py", line 222, in trigger
    self.recvAck()
  File "/home/pi/photobooth/photobooth/Photobooth.py", line 112, in recvAck
    if self.recvEvent(events) != 0:
  File "/home/pi/photobooth/photobooth/Photobooth.py", line 104, in recvEvent
    raise ValueError('Unknown event received', str(event))
ValueError: ('Unknown event received', 'triggered')

```

Maybe the current which arrives at the pin while triggering is to low/not long enough (resisstance of Thrones cable to high)? 
If I press the button for a few seconds instead of hitting it for a short moment, it seems like the chance of taking a  picture is higher.

I also got some other warnings/errors which would be nice to fix, but it seems like they don't stop the photobooth working.

Thank you in advance!

-Axel

Edit:
removed serial Numbers
Corrected camera model

**Comment by wikijm** [comment]
Hi @axel8417,

2 things to take care first:
- You leave your camera serial number in logs, which can be considerate as a sensitive data. You should think of delete it to your issue
- You are talking about 500D model when I see in your logs a reference of 600D model. I assume that logs are more accurate 👼 

For your main issue, could you please try to trigger `gphoto2` only, with your buzzer ?
This way, we will be able to see which errors are coming from gphoto2 and which are coming from `reuterbal/photobooth`

**Comment by reuterbal** [comment]
Hi @axel8417,

first of all: thanks for trying out the software and taking the time to report back issues!

Secondly, your Logfile output and the console output don't seem to match each other. There is a difference of 7 seconds in the time stamps. However, I think I know the reason of the issues:

As everything works fine when you use short cables, this looks like an electrical engineering issue and a number of electrical effects interplaying here:
1. Make sure that your button closes the GPIO pin 23 and GND (**not** +3.3V).
2. If you use such long wires, keep in mind that they have a non-neglectable electrical resistance, especially for such low voltages (3.3 V for GPIO) - thus, use wires with a large cross-sectional area to reduce the resistance. If the resistance of the wire is high, this leads to a considerable voltage drop and might affect the PIs ability to detect a button press (because the actual voltage drop might be of the same order than the potential difference). Read about this e.g. [here](https://en.wikipedia.org/wiki/Voltage_drop) and [here](https://en.wikipedia.org/wiki/Electrical_resistance_and_conductance).
3. Depending on your button type, it can generate some bouncing in the electrical signal (meaning, there is not a clean voltage drop pulling the GPIO pin to GND but instead some oscillations in the signal). For most cases this is sufficiently counteracted by the internal pullup resistor of the Pi, however, for such long wiring this might not suffice and you might need an external resistor in series.

The above leads to detection of a second button press, while the software did not yet start handling the first. In the trigger()-function the detection of a button press is disabled to avoid such problems, however the short time frame between the first press and the disabling seems to be too long to effectively avoid these problems.

I changed the order in the trigger()-function to disable the trigger button first and initiate the GUI output afterwards. This might help reduce the problems, however there is a better way to fix this which I might do in the next couple of days (I want to change some things with respect to GPIO anyways).

You should do the following:
1. Use wires with a larger cross-sectional area to reduce the resistance. This should help detecting the button press in the first place.
2. Try out the latest version to see if it helped to avoid the freezing.

**Comment by axel8417** [comment]
Hi,
I think too that it is an electrical issue. I have tested it a little today:

1. Sure
2. I am using a normal 230V cable with ~ 0,5mm² cross-sectual area, which should be enough?
3. I removed the button and just connected the two wires

I also tested the latest version.
Unfortunatelly it still does mit work.
I will try a 1,5mm² cable on friday.
I know that there is a setting which changes the sensitivity of the gpio pin. Maybe playing with it would also work?
Should I try powering the button with an external power supply to avoid the voltage drop?
I think it is strange that in V2 the photobooth does not trigger correct (voltage drop, because of high resistance), but in V1 it was triggering randomly (maybe electrical induction -> low resistance after putting in an additional resistance it improved?). Pleased correct me If I am wrong. I am Just wondering why both Versions do not work correctly.
